Public/js/profile-app.js

```diff
@@ -121,12 +123,83 @@ $(document).ready(function() {
       html +=   '<td>';
       html +=     day;
       html +=   '</td>';
-      html +=   '<td><button class="btn btn-success">Remove</button></td>';
+      html +=   '<td><button class="btn btn-success remove">Remove</button></td>';
       html += '</tr>'
       $('#table').append(html);
     }
   };
 
+  Reservation.prototype.runRemove = function(button) {
+    $.ajax({
+      context: this,
+      type: 'DELETE',
+      url: 'reservations',
+      data: {
+        reservation: {
+          machine: this.machine,
+          hour: this.hour,
+          day: this.day
+        }
+      },
+      dataType: 'json',
+      success: function(response) {
+        console.log(response);
+        if(response.ok) {
+          $(button).parent().parent().remove();
+        } else if(response.authenticated == false) {
+          logOut();
+        } else {
+          console.log(response);
+        }
+      }
+    });
+  };
+
+  User.prototype.updateUser = function() {
+    console.log(this)
+    $.ajax({
+      context: this,
+      type: 'PUT',
+      url: 'users',
+      data: {
+        updatedUser: {
+          oldPassword : this.oldPassword,
+          newPassword : this.newPassword,
+          email : this.email
+        }
+      },
+      dataType: 'json',
+      success: function(response) {
+        $('#response').remove();
+        if(response.ok) {
+          $('#change-password .modal-body').append('<p id="response" style="color:green;">Password successfully changed</p>');
+          $('#change-email .modal-body').append('<p id="response" style="color:green;">Email successfully changed</p>');
+          if(this.email) {$('#current-email').text(this.email);}
+          setTimeout(function() {
```

Comment:
Just a style thing, to let the user know that there change password/email request has worked.